File: src/CSharp/PolyData/ColorDisconnectedRegions.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Windows.Forms;
using System.Diagnostics;

using Kitware.VTK;

namespace ActiViz.Examples {
   public partial class Form1 : Form {
      public Form1() {
         InitializeComponent();
      }


      private void renderWindowControl1_Load(object sender, EventArgs e) {
         try {
            ColorDisconnectedRegions();
         }
         catch(Exception ex) {
            MessageBox.Show(ex.Message, "Exception", MessageBoxButtons.OK);
         }
      }


      private void ColorDisconnectedRegions() {
         // Create some spheres
         vtkSphereSource sphereSource1 = vtkSphereSource.New();
         sphereSource1.Update();

         vtkSphereSource sphereSource2 = vtkSphereSource.New();
         sphereSource2.SetCenter(5, 0, 0);
         sphereSource2.Update();

         vtkSphereSource sphereSource3 = vtkSphereSource.New();
         sphereSource3.SetCenter(10, 0, 0);
         sphereSource3.Update();

         vtkAppendPolyData appendFilter = vtkAppendPolyData.New();
         appendFilter.AddInputConnection(sphereSource1.GetOutputPort());
         appendFilter.AddInputConnection(sphereSource2.GetOutputPort());
         appendFilter.AddInputConnection(sphereSource3.GetOutputPort());

         vtkPolyDataConnectivityFilter connectivityFilter = vtkPolyDataConnectivityFilter.New();
         connectivityFilter.SetInputConnection(appendFilter.GetOutputPort());
         connectivityFilter.SetExtractionModeToAllRegions();
         connectivityFilter.ColorRegionsOn();
         connectivityFilter.Update();

         // Visualize
         vtkPolyDataMapper mapper = vtkPolyDataMapper.New();
         mapper.SetInputConnection(connectivityFilter.GetOutputPort());
         double[] range = connectivityFilter.GetOutput().GetPointData().GetArray("RegionId").GetRange();
         mapper.SetScalarRange(range[0], range[1] );

         vtkActor actor = vtkActor.New();
         actor.SetMapper(mapper);
         // get a reference to the renderwindow of our renderWindowControl1
         vtkRenderWindow renderWindow = renderWindowControl1.RenderWindow;
         // renderer
         vtkRenderer renderer = renderWindow.GetRenderers().GetFirstRenderer();
         // set background color
         renderer.SetBackground(0.0, 0.0, 0.0);
         // add our actor to the renderer
         renderer.AddActor(actor);
      }
   }
}

File: src/Java/PolyData/ColorDisconnectedRegions.java
import vtk.vtkActor;
import vtk.vtkAppendPolyData;
import vtk.vtkNativeLibrary;
import vtk.vtkPolyDataConnectivityFilter;
import vtk.vtkPolyDataMapper;
import vtk.vtkRenderer;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkSphereSource;


public class ColorDisconnectedRegions {
  private static final long serialVersionUID = 1L;


  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) {
        if (!lib.IsLoaded()) {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------


  public static void main(String s[]) {
    // Create some spheres
    vtkSphereSource sphere1 = new vtkSphereSource();
    sphere1.Update();

    vtkSphereSource sphere2 = new vtkSphereSource();
    sphere2.SetCenter(5.0, 0.0, 0.0);
    sphere2.Update();

    vtkSphereSource sphere3 = new vtkSphereSource();
    sphere3.SetCenter(10.0, 0.0, 0.0);
    sphere3.Update();

    vtkAppendPolyData appendFilter = new vtkAppendPolyData();
    appendFilter.AddInputConnection(sphere1.GetOutputPort());
    appendFilter.AddInputConnection(sphere2.GetOutputPort());
    appendFilter.AddInputConnection(sphere3.GetOutputPort());
    appendFilter.Update();

    vtkPolyDataConnectivityFilter connectivityFilter = new vtkPolyDataConnectivityFilter();
    connectivityFilter.SetInputConnection(appendFilter.GetOutputPort());
    connectivityFilter.SetExtractionModeToAllRegions();
    connectivityFilter.ColorRegionsOn();
    connectivityFilter.Update();

    //Create a Mapper and Actor
    vtkPolyDataMapper mapper = new vtkPolyDataMapper();
    mapper.SetInputConnection(connectivityFilter.GetOutputPort());
    mapper.SetScalarRange(connectivityFilter.GetOutput().GetPointData().GetArray("RegionId").GetRange());
    mapper.Update();

    vtkActor actor = new vtkActor();
    actor.SetMapper(mapper);

    // Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);

    // Visualise the arrow
    ren.AddActor(actor);
    ren.ResetCamera();
    ren.GetActiveCamera().Zoom(0.9);

    renWin.SetSize(600, 600);
    renWin.Render();

    iren.Initialize();
    iren.Start();

  }
}
